Tools/src/FormatCheckTool/FormatCheckTool.cs

```diff
@@ -0,0 +1,234 @@
+
+            ShowIntro();
+
+            while (true)
+            {
+                Console.WriteLine("Enter the specs JSON file path to be checked:");
+                var input = Console.ReadLine()?.Trim();
+                Console.WriteLine();
+
+                if (input?.ToLower(CultureInfo.InvariantCulture) == "exit")
+                {
+                    break;
+                }
+
+                // Validate input
+                if (input?.Length > 0)
+                {
+                    var pathSplit = input.Split(Path.DirectorySeparatorChar);
+                    string basicDir = pathSplit.Length > 1 ? string.Join(Path.DirectorySeparatorChar, pathSplit.Take(pathSplit.Length - 1)) : $".{Path.DirectorySeparatorChar}";
+                    string basicPath = pathSplit[pathSplit.Length - 1];
+                    if (!Directory.Exists(basicDir)) {
+                        Console.WriteLine($"File Path Error. {PleaseCheckStr}");
+                        continue;
+                    }
+
+                    string[] paths = Directory.GetFiles(basicDir, basicPath);
+                    Console.WriteLine($"Loading {paths.Length} Spec Files...");
+                    Console.WriteLine();
+
+                    foreach (string path in paths)
+                    {
+                        CheckPipeline(path);
+                    }
+
+                    Console.WriteLine();
+                }
+            }
+        }
+
+        private static void CheckPipeline(string filePath)
+        {
+            Console.WriteLine("-------------------------");
+            Console.WriteLine($"Begin to check file {filePath}.");
+            Tuple<bool, List<TestModel>> data = LoadJsonFile(filePath);
+            if (!data.Item1)
+            {
+                return;
+            }
+
+            bool stage2 = CheckBasicContent(data.Item2);
+            if (!stage2)
+            {
+                return;
+            }
+
+            bool stage3 = CheckModelResult(data.Item2);
+        }
+
+        private static Tuple<bool, List<TestModel>> LoadJsonFile(string filePath)
+        {
+            List<TestModel> specs = new List<TestModel>{ };
+            string flag = string.Empty;
+            if (!File.Exists(filePath))
+            {
+                flag = $"File {filePath} Dose Not Exist! {PleaseCheckStr}";
+            }
+            else
+            {
+                var rawData = File.ReadAllText(filePath);
+
+                // Try to load Specs JSON file, and catch exception.
+                try
+                {
+                    specs = (List<TestModel>)JsonConvert.DeserializeObject<IList<TestModel>>(rawData);
+                    if (!specs.Any())
+                    {
+                        flag = $"JSON File is Empty, {PleaseCheckStr}";
+                    }
+                }
+                catch (JsonReaderException jre)
+                {
+                    flag = $"JSON Reader Exception! {PleaseCheckStr}";
```

Comment:
For any exceptions, it would be better to log also some more details. At least the exception message/type and where it happened.